test/Passes/tile-and-fuse-default.mlir

```diff
@@ -0,0 +1,238 @@
+      outs(%arg2 : tensor<4x4x32x32xf32>) {
+    ^bb0(%in: f32, %in_2: f32, %out: f32):
+      %1 = arith.mulf %in, %in_2 : f32
+      %2 = arith.addf %out, %1 : f32
+      linalg.yield %2 : f32
+  } -> tensor<4x4x32x32xf32>
+  %3 = linalg.generic {
+      indexing_maps = [#map3],
+      iterator_types = ["parallel", "parallel", "parallel", "parallel"]}
+      outs(%0 : tensor<4x4x32x32xf32>) {
+    ^bb0(%out: f32):
+      %4 = arith.maxf %out, %c0 : f32
+      linalg.yield %4 : f32
+  } -> tensor<4x4x32x32xf32>
+  return %3 : tensor<4x4x32x32xf32>
+}
+
+// CHECK: #[[MAP:.+]] = affine_map<(d0, d1, d2, d3) -> (d0, d1, d3)>
+// CHECK: #[[MAP1:.+]] = affine_map<(d0, d1, d2, d3) -> (d0, d3, d2)>
+// CHECK: #[[MAP2:.+]] = affine_map<(d0, d1, d2, d3) -> (d1, d2)>
+// CHECK: #[[MAP3:.+]] = affine_map<(d0, d1) -> (d0, d1)>
+// CHECK: func.func @matmul_eletwise(
+// CHECK-DAG: %[[C4:.+]] = arith.constant 4 : index
+// CHECK-DAG: %[[C0:.+]] = arith.constant 0 : index
+// CHECK-DAG: %[[C1:.+]] = arith.constant 1 : index
+// CHECK: %[[LOOP:.+]] = scf.for %{{.+}} = %[[C0]] to %[[C4]] step %[[C1]]
+// CHECK-NEXT: %[[LOOP1:.+]] = scf.for %{{.+}} = %[[C0]] to %[[C4]] step %[[C1]]
+// CHECK-COUNT-2: linalg.generic
+// CHECK: scf.yield %{{.+}} : tensor<4x4x32x32xf32>
+// CHECK-NEXT: }
+// CHECK: scf.yield %{{.+}} : tensor<4x4x32x32xf32>
+// CHECK-NEXT: }
+
+// -----
+
+#map = affine_map<(d0, d1) -> (d0, d1)>
+
+func.func @matmul_sequence_fusion(%arg0: tensor<32x64xf32>, %arg1: tensor<64x32xf32>,
+    %arg2: tensor<32x32xf32>, %arg3: tensor<32x64xf32>, %arg4: tensor<32x64xf32>,
+    %arg5: tensor<64x32xf32>, %arg6: tensor<32x32xf32>) -> tensor<32x32xf32> {
+  %c0 = arith.constant 0.0 : f32
+  %0 = linalg.matmul ins(%arg0, %arg1 : tensor<32x64xf32>, tensor<64x32xf32>)
+    outs(%arg2 : tensor<32x32xf32>) -> tensor<32x32xf32> // [M, N0] * [N0, N1]
+  %1 = linalg.matmul ins(%0, %arg3 : tensor<32x32xf32>, tensor<32x64xf32>)
+    outs(%arg4 : tensor<32x64xf32>) -> tensor<32x64xf32> // [M, N1] * [N1, N2]
+  %2 = linalg.matmul ins(%1, %arg5 : tensor<32x64xf32>, tensor<64x32xf32>)
+    outs(%arg6 : tensor<32x32xf32>) -> tensor<32x32xf32> // [M, N2] * [N2, N3]
+  %3 = linalg.generic {indexing_maps = [#map],
+                       iterator_types = ["parallel", "parallel"]}
+    outs(%2: tensor<32x32xf32>) {
+      ^bb0(%out: f32):
+        %4 = arith.maxf %out, %c0 : f32
+        linalg.yield %4 : f32
+  } -> tensor<32x32xf32>
+  return %3 : tensor<32x32xf32>
+}
+
+// CHECK-DAG: #[[MAP:.+]] = affine_map<() -> ()>
+// CHECK: func.func @matmul_sequence_fusion
+// CHECK-DAG: %[[C32:.+]] = arith.constant 32 : index
+// CHECK-DAG: %[[C0:.+]] = arith.constant 0 : index
+// CHECK-DAG: %[[C1:.+]] = arith.constant 1 : index
+// CHECK: %[[LOOP:.+]] = scf.for %{{.+}} = %[[C0]] to %[[C32]] step %[[C1]]
+// CHECK-NEXT: %[[LOOP1:.+]] = scf.for %{{.+}} = %[[C0]] to %[[C32]] step %[[C1]]
+// CHECK-COUNT-3: linalg.matmul
```

Comment:
What are the types for these linalg ops?

Reply:
the generic are scalar here. Basically we have:
```
for ( M
  for (N
    linalg.matmul (only on the reduction loop)
    linalg.generic (scalar op)
```

Reply:
Can you just add the types to the CHECK lines, to make it clear what's going on?

I'm guessing `CHECK-COUNT` can cope with regexes, but if not, split into one with regex and the rest as a COUNT.